locoprop/layer.py

```diff
@@ -1,19 +1,106 @@
+import dataclasses
+from typing import Optional, Any
+
+import torch
 from torch import nn
 
+_IS_TRAINING = [False]
```

Comment:
What's the reason this needs to be a list? Couldn't it be just a boolean? Also, is this already compatible with MP? If not, could we easily make it so?

Reply:
I didn't figure out how to have one module set a boolean and the other read it. Using a list, it becomes trivial. Do you have any idea to solve this?

Reply:
It's technically not compatible with multiprocessing or model-parallel. Practically, as long as all python processes are synchronised, it should be fine. (Synchronisation is a requirement anymore for torch model-parallel)

Reply:
I see, should be good then. Maybe we can add a comment about this:
```suggestion
# Using a list to enable read/write operations by different modules.
# Not strictly multiprocessing-compatible, expect unexpected behavior.
_IS_TRAINING = [False]
```